### ISSUE TITLE
rule suggestion: `no-import-meta-env`

### ISSUE THREAD

**Issue by LBBO** [oxc-project/oxc#22372]
Similarly to the [`node/no-process-env`](https://oxc.rs/docs/guide/usage/linter/rules/node/no-process-env) rule which forbids the use of `process.env`, I think it makes sense to have a rule that forbids the use of `import.meta.env`. I'd even be fine with it being an option of the existing `node/no-process-env` rule, since they basically fulfill the same purpose.

If this is indeed a desired feature, I'd be happy to contribute a PR introducing it. Just let me know whether it should be its own rule or a feature of the existing rule.

**Comment by camc314** [comment]
Hi, thanks for the suggestion!

This isn't planned at present. Please use a custom JS plugin:


```ts
// `plugin.ts`
import type { Plugin } from "oxlint/plugins";

const plugin: Plugin = {
  meta: {
    name: "my-plugin",
  },
  rules: {
    "no-import-meta-env": {
      create(context) {
        return {
          MetaProperty(meta) {
            // disallow `import.meta.env`

            if (
              meta.meta.name === "import" &&
              meta.property.name === "meta" &&
              meta.parent &&
              meta.parent.type === "MemberExpression" &&
              meta.parent.property.type === "Identifier" &&
              meta.parent.property.name === "env"
            ) {
              context.report({
                node: meta,
                message: "Using import.meta.env is not allowed.",
              });
            }
          },
        };
      },
    },
  },
};

export default plugin;

```

```json
// `.oxlintrc.json`
{
  "jsPlugins": ["./plugin.ts"],
  "rules": {
    "my-plugin/no-import-meta-env": "error"
  }
}
```



If you have any issues, please reach out - thanks!
